### DISCUSSION TITLE
How do I install custom certificate authorities for use with brew

### DISCUSSION THREAD

**Discussion by seanthegeek** [Homebrew/discussions#4262] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.20-184-gcdf58fe
ORIGIN: https://github.com/Homebrew/brew
HEAD: cdf58fe392beea957cb0a77007d5308aee3e336f
Last commit: 2 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 9ad93a700802b48d4e972e9e8802187ba48f73b2
Core tap last commit: 7 days ago
Core tap branch: master
Core tap JSON: 02 Feb 07:28
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_INSTALL_FROM_API: set
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.6.2-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
```

### Output of `brew doctor`

```shell
lease note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libsbuspkcs11.dylib

Warning: You have uncommitted modifications to Homebrew/homebrew-core.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core && git stash -u && git clean -d -f

Uncommitted files:
  UU Formula/protobuf.rb
```

### Description of issue

We have SSL inspection enabled on our corporate firewalls, and get errors when trying to download brew packages. It looks like brew isn't using our custom Certificate Authority (CA) in the macOS key store. How can we add a CA to the key store brew is using?

**Comment by alebcay** [comment]
Under the hood, `brew` uses `curl` for most network operations. To get `brew` to trust a custom CA is (at least for the most part) to get `curl` to trust a custom CA.
- Create a curlrc file to configure `curl` to trust your custom CA. [This can probably be done with the `--cacert` flag on `curl`](https://serverfault.com/a/643765). You can refer to [the `curl` manual](https://curl.se/docs/manual.html#config-file) for details on how to express this flag in the curlrc file, or [a sample curlrc file to get an idea of what the file might look like](https://gist.github.com/v1m/f1d4751883f19c916515).
- Tell `brew` to use the curlrc file by setting [the environment variable `HOMEBREW_CURLRC`](https://docs.brew.sh/Manpage#environment). By default, `brew` runs `curl` with the `--disable` flag which prevents curlrc files from having any effect; by setting this environment variable, this prevents the `--disable` flag from being added, and allows your curlrc file to take effect.

Note that I haven't tried any of these steps, but it should get you moving in the right direction.
